javascripts/services/ContactService.js

```diff
@@ -0,0 +1,43 @@
+"use strict";
+
+app.service("ContactService", function($http, $q, FIREBASE_CONFIG) {
+		const getViewedContacts= (userUid) => {
+			 let contacts = [];
+			 return $q((resolve, reject) => {
+			 	$http.get(`${FIREBASE_CONFIG.databaseURL}/contacts.json?orderBy="uid"&equalTo="${userUid}"`).then((results) => {
+			 		let fbContacts = results.data;
+			 		
+			 		Object.keys(fbContacts).forEach((key) => {
+	                    fbContacts[key].id = key; 
+	                    contacts.push(fbContacts[key]);
+	                	resolve(contacts);
```

Comment:
make sure to resolve your contacts outside of your forEach loop going over the Object.keys